protocol/messenger_communities.go

```diff
@@ -5787,44 +5765,38 @@ func (m *Messenger) startCommunityRekeyLoop() {
 	}()
 }
 
-// rekeyAllCommunities attempts to rekey every community in persistence.
-// A community will be rekeyed if it meets all the following criteria:
-//   - Community.IsAdmin()
-//   - Community.Encrypted()
-//   - Community.RekeyedAt().Add(rki).Before(time.Now()) where rki is a defined rekey interval
-func (m *Messenger) rekeyAllCommunities(logger *zap.Logger) {
-	// Determine the rekey interval, if the value is not set as a property of m.communitiesManager
-	// default to one hour
+// rekeyCommunities loops over controlled communities and rekeys if rekey interval elapsed
+func (m *Messenger) rekeyCommunities(logger *zap.Logger) {
 	// TODO in future have a community level rki rather than a global rki
-	/*
-		var rki time.Duration
-		if m.communitiesManager.RekeyInterval == 0 {
-			rki = time.Hour
-		} else {
-			rki = m.communitiesManager.RekeyInterval
-		}*/
+	var rekeyInterval time.Duration
+	if m.communitiesManager.RekeyInterval == 0 {
+		rekeyInterval = 48 * time.Hour
+	} else {
+		rekeyInterval = m.communitiesManager.RekeyInterval
+	}
 
-	// Get and loop over all communities in persistence
-	cs, err := m.Communities()
+	controlledCommunities, err := m.ControlledCommunities()
 	if err != nil {
 		logger.Error("error getting communities", zap.Error(err))
 		return
 	}
-	for _, c := range cs {
+	for _, c := range controlledCommunities {
+		key, err := m.sender.GetCurrentKeyForGroup(c.ID())
```

Comment:
not sure if that's the best way, maybe is safer to just check if the community is Encrypted?
It should not probably happen, but say that for whatever reason a key is not generated, if we check `Encrypted` it would self heal, while if we check whether there's already a key present, this will not trigger.
But I saw that you removed the Encrypted check, maybe there's a reason why we would want to use getCurrent... (would it not though rekey also non encrypted communities, in case encryption is toggled off?)

Reply:
That's a good point. Updated, please recheck, also added handling for channels re-keying.